comp/otelcol/collector/impl/collector.go

```diff
@@ -81,8 +84,79 @@ func (c *converterFactory) Create(_ confmap.ConverterSettings) confmap.Converter
 	return c.converter
 }
 
+func newConfigProviderSettings(reqs Requires, enhanced bool) otelcol.ConfigProviderSettings {
+	converterFactories := []confmap.ConverterFactory{
+		expandconverter.NewFactory(),
+	}
+
+	if enhanced {
+		converterFactories = []confmap.ConverterFactory{
```

Comment:
nit: I would probably just append to the `converterFactories` slice, avoid unnecessary additional allocations.